flake.nix

```diff
@@ -0,0 +1,117 @@
+{
+  description = "Hyperspace";
+
+  inputs.nixpkgs.url = "nixpkgs/nixpkgs-unstable";
+
+  outputs = { self, nixpkgs }:
+    let
+      supportedSystems =
+        [ "x86_64-linux" "x86_64-darwin" "i686-linux" "aarch64-linux" ];
+      forAllSystems = f:
+        nixpkgs.lib.genAttrs supportedSystems (system: f system);
+      packageJson = builtins.fromJSON (builtins.readFile ./package.json);
+
+    in {
+      overlay = final: prev: {
+        hyperspace = final.mkYarnPackage {
+          name = packageJson.name;
+          src = ./.;
+          packageJSON = ./package.json;
+          yarnLock = ./yarn2nix.lock;
+        };
+      };
+
+      defaultPackage = forAllSystems (system:
+        (import nixpkgs {
+          inherit system;
+          overlays = [ self.overlay ];
+        }).hyperspace);
+
+      apps = forAllSystems (system: {
+        refresh-dependencies = let pkgs = import nixpkgs { inherit system; };
+        in {
+          type = "app";
+          program = "${
+              pkgs.writeShellScriptBin "refresh-dependencies" ''
```

Comment:
Perhaps we could use [`writeShellScript`](https://github.com/NixOS/nixpkgs/blob/3e0ce8c5d478d06b37a4faa7a4cc8642c6bb97de/pkgs/build-support/trivial-builders.nix#L203-L226) here?